### DISCUSSION TITLE
using radix with remix

### DISCUSSION THREAD

**Discussion by MadadiSaeed** [radix-ui/primitives#2106] [Help] [answered]
I am using radix dropdown-menu in my remix app like this:
```
<DropdownMenu.Root>
  <DropdownMenu.Trigger>
    <p>trigger</p>
  </DropdownMenu.Trigger>
  <DropdownMenu.Content>
    <div>
      <p>content section item</p>
      <p>content section item</p>
      <p>content section item</p>
      <p>content section item</p>
    </div>
  </DropdownMenu.Content>
</DropdownMenu.Root>
```
But the problem is that after opening dropdown, radix adds a bunch of additional attributes and elements to my app. This is true for select component too. **Why is that**?
**Before opening Dropdown**:
![Screenshot from 2023-04-24 19-39-46](https://user-images.githubusercontent.com/18693428/234057058-ea94e365-ef2b-4aea-aff0-c02a7f6c5a5d.png)
**After opening Dropdown**:
![Screenshot from 2023-04-24 19-39-51](https://user-images.githubusercontent.com/18693428/234057636-115e7619-95eb-4f1d-a973-c693ef65884c.png)

**Comment by benoitgrelard** [comment, marked as answer]
These are related to modality and accessibility, making sure that whatever is outside of the menu isn't visible to screen readers or interactive, focus is trapped, etc.

That shouldn't be anything to worry about though. Are you having specific issues?
